public/javascripts/Choropleth.js

```diff
@@ -1,241 +1,327 @@
-                distanceLeft = "0";
-            }
-            else if (milesLeft === 0){
-                distanceLeft = "<1";
-            }
-            else if (milesLeft === 1){
-                distanceLeft = "1";
-            }
-            else{
-                distanceLeft = ">1";
+    /**
+     * This function finds the color for a specific region of the accessibility choropleth.
+     * 
+     * @param {*} rate Object from which information about labels is retrieved.
+     */
+    function findAccessibilityChoroplethColor(rate) {
+        var totalIssues = 0;
+        for (var issue in rate.labels) {
+            if (rate.labels.hasOwnProperty(issue)) {
+                totalIssues += rate.labels[issue];
             }
+        }
+        var significantData = rate.rate >= .3;
+        var fillColor = significantData ? getColor(1000.0 * totalIssues / rate.completed_distance_m) : '#888';
+        var fillOpacity = significantData ? 0.4 + (totalIssues / rate.completed_distance_m) : .25;
+        return {
+            color: '#888',
+            weight: 1,
+            opacity: 0.25,
+            fillColor: fillColor,
+            fillOpacity: fillOpacity
+        }
+    }
 
-            var data = "Click_module=Choropleth_regionId="+regionId+"_distanceLeft="+distanceLeft+"_target=audit";
-            postToWebpageActivity(data);
-        });
+    /**
+     * This function sets the specific popup content of each region of the accessibility choropleth.
+     * 
+     * @param {*} rate Object from which information about labels is retrieved.
+     */
+    function setAccessibilityChoroplethPopupContent(rate) {
+        var labels = rate.labels;
+        var counts = {};
+        for (var j in labelText) {
+            if (typeof labels[j] != 'undefined')
+                counts[j] = labels[j];
+            else
+                counts[j] = 0;
+        }
+        return '<div class="resultsImages"><table><tbody>'+
+               '<tr><td>' + i18next.t('missing-sidewalks') + '<br/></td>'+
+               '<td>' + i18next.t('missing-ramps') + '<br/></td>'+
+               '<td>' + i18next.t('surface-problems') + '<br/>'+
+               '<td>' + i18next.t('sidewalk-obstacles') + '<br/></td></td></tr>' +
+               '<tr><td><img src="/assets/javascripts/SVLabel/img/cursors/Cursor_NoSidewalk.png"></td>'+
+               '<td><img src="/assets/javascripts/SVLabel/img/cursors/Cursor_NoCurbRamp.png"></td>'+
+               '<td><img src="/assets/javascripts/SVLabel/img/cursors/Cursor_SurfaceProblem.png"></td>'+
+               '<td><img src="/assets/javascripts/SVLabel/img/cursors/Cursor_Obstacle.png"></td>'+
+               '<tr><td>'+ counts['NoSidewalk'] +'</td><td>'+ counts['NoCurbRamp'] +'</td><td>'+ counts['SurfaceProblem'] +'</td><td>'+ counts['Obstacle'] +'</td></tr></tbody></table></div>';    
     }
 
+    if (params.choroplethType === 'results') {
+        $.getJSON('/adminapi/choroplethCounts', function (labelCounts) {
```

Comment:
When we pass the issue counts to `initializeChoropleth`, let's actually split the issue counts into it's own parameter instead of having it directly connected to the polygon data. with the current method, we just have to know by context whether or not to expect a `labels` object associated with each polygon. Instead, it should be separate. Then in either the `initializeChoropleth()` or `initializeChoroplethNeighborhoodPolygons()` function, we can check if we are loading the results choropleth and the issue count data are missing, log an error to the console or something.

Reply:
I'm a little confused about why you are saying log an errror to the console.

Reply:
I'm saying that there are two things that both need to get passed for the results choropleth to load correctly. We need the `choroplethType` parameter set to 'results' and we need the label count data passed. If the `choroplethType` is set to 'results' we shouldn't just assume that the label count data is also passed, we should check if it is and log an error if it isn't.

We just want to avoid making too many assumptions about what data we do/don't have. And if someone forgets to include something, it would be nice for them to see a message saying what they left out.

Reply:
Ok, I think I understand what you mean.  I just pushed some changes that address this issue.